cpp/ql/src/Diagnostics/FailedExtractorInvocations.ql

```diff
@@ -1,7 +1,7 @@
 /**
  * @name Failed extractor invocations
  * @description Gives the command line of compilations for which extraction did not run to completion.
- * @kind diagnostic
+ * @kind table
```

Comment:
Do you know what `@kind diagnostic` is supposed to be?  Some of them seem to have a 'severity' value in the third column of the `select`, while others don't have a third column.

Reply:
For diagnostic queries, all of the following result patterns are accepted:
- `message, <optional list of placeholder pairs>`
- `message, <optional list of placeholder pairs>, severity`
- `entity, message, <optional list of placeholder pairs>` (like a regular alert query)
- `entity, message, <optional list of placeholder pairs>, severity`

Looking at the warning you're seeing, if an `entity` is provided we do need it to have a location, and in this case `Compilation` is not `Locatable`. If you want to keep this query as `@kind diagnostic`, you could either use an entity with a location (if a suitable one exists), or use one of the first two formats in the list above.